### PR TITLE
Allow students to have multiple questions across multiple courses

### LINKED ISSUE #845: Allow students to have multiple questions across multiple courses
From a bug report:

> I was in one quese for two hours and wanted to join a queue for a different class for clarification and the other question is deleted.

The student was really frustrated, for good reason. We should allow students to have multiple questions across courses simultaneously.

### PR DESCRIPTION
**Pull request by tiingweii-shii**
# Description
Allows one question per course for each student.

Previously, only one question is allowed per user. We believe it is more ideal to allow a student to have one question per course. This PR also updated the warning for when a student wants to create a second question for a course.

Closes #845 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe how you tested this PR (both manually and with tests)
Provide instructions so we can reproduce. 

- [x] This has been manually tested in the front end. (In database: Create a second course in course_model. Add a student to the second course by adding an entry in user_course_model. On the app: Sign in as the student. Create a question in one of the courses that the student is in. Then create another question in the other course. This should not generate a warning. However, creating two questions in one course will give a warning.
- [x] One of the test in `question.controller.ts` has been adjusted to reflect that a second question in the same course (in different queues) gives a 400. A test has been added to show that a user can have one question each for two different courses.
- [x] This also has been tested with courses with more than one queues.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code where needed 
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by VKong6019 — APPROVED** [review]
yeehaw

File: packages/app/components/Queue/Student/StudentQueue.tsx [hunk 1]
```diff
@@ -292,7 +296,13 @@ export default function StudentQueue({
             buttons={
               !studentQuestion && (
                 <Popconfirm
-                  title="In order to join this queue, you must delete your previous question. Do you want to continue?"
+                  title={
+                    <PopConfirmTitle>
+                      You already have a question in a queue for this course, so
```
Comment (dfarooq610) [inline packages/app/components/Queue/Student/StudentQueue.tsx:301]: So we allow them to have multiple questions across the different courses. Do we still need this popup? 

Can't we just disable the button altogether? if they have a question in this course, then they can just edit their existing question. 

I think we can Disable the button and add a tooltip indicating that they can only have 1 question in the queue at a time

Comment (tiingweii-shii) [reply packages/app/components/Queue/Student/StudentQueue.tsx:301]: Oh so we were thinking about hybrid queues when we implemented this.
Maybe there are students who want to leave an in-person queue for whatever reason and join an online queue instead (or vice versa)?

Comment (dfarooq610) [reply packages/app/components/Queue/Student/StudentQueue.tsx:301]: Yea that makes sense, we can keep it in then

**Review by dfarooq610 — APPROVED** [review]

**Pull request merged**